lib/ts/recipe/recipeModule/types.ts

```diff
@@ -12,20 +12,28 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-import { NormalisedAppInfo, PostAPIHookFunction, RecipePreAPIHookContext } from "../../types";
+import { NormalisedAppInfo, RecipePostAPIHookContext, RecipePreAPIHookContext } from "../../types";
 
-export type RecipeConfig<Action, PreAPIHookContext extends RecipePreAPIHookContext<Action>> = {
+export type RecipeConfig<
+    Action,
+    PreAPIHookContext extends RecipePreAPIHookContext<Action>,
+    PostAPIHookContext extends RecipePostAPIHookContext<Action>
+> = {
     recipeId: string;
     appInfo: NormalisedAppInfo;
     preAPIHook?: (context: PreAPIHookContext) => Promise<{ url: string; requestInit: RequestInit }>;
-    postAPIHook?: PostAPIHookFunction;
+    postAPIHook?: (context: PostAPIHookContext) => Promise<void>;
 };
 
-export type NormalisedRecipeConfig<Action, PreAPIHookContext extends RecipePreAPIHookContext<Action>> = {
+export type NormalisedRecipeConfig<
+    Action,
+    PreAPIHookContext extends RecipePreAPIHookContext<Action>,
+    PostAPIHookContext extends RecipePostAPIHookContext<Action>
```

Comment:
You don't need to define these generics too. It can just be like:
```ts
export type NormalisedRecipeConfig<
    Action
> = {
    recipeId: string;
    appInfo: NormalisedAppInfo;
    preAPIHook: (context: RecipePreAPIHookContext<Action>) => Promise<{ url: string; requestInit: RequestInit }>;
    postAPIHook: (context: RecipePostAPIHookContext<Action>) => Promise<void>;
};
```

Reply:
Same goes for `RecipeConfig` above

Reply:
Done, changed to `RecipeConfig<Action>` and `NormalisedRecipeConfig<Action>`